cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationHelper.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.resource;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.api.model.apps.StatefulSet;
+import io.strimzi.api.kafka.model.KafkaResources;
+import io.strimzi.operator.cluster.ClusterOperator;
+import io.strimzi.operator.cluster.model.KafkaCluster;
+import io.strimzi.operator.common.AdminClientProvider;
+import io.strimzi.operator.common.Util;
+import io.strimzi.operator.common.model.Labels;
+import io.strimzi.operator.common.operator.resource.SecretOperator;
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.Config;
+import org.apache.kafka.clients.admin.DescribeConfigsResult;
+import org.apache.kafka.common.config.ConfigResource;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * This class contains methods for getting current configuration from the kafka brokers asynchronously.
+ */
+public class KafkaBrokerConfigurationHelper {
+
+    private final AdminClientProvider adminClientProvider;
+    protected final SecretOperator secretOperations;
+    private static final Logger log = LogManager.getLogger(KafkaBrokerConfigurationHelper.class);
+
+    public KafkaBrokerConfigurationHelper(AdminClientProvider adminClientProvider, SecretOperator secretOperations) {
+        this.adminClientProvider = adminClientProvider;
+        this.secretOperations = secretOperations;
+    }
+
+    public Future<Admin> adminClient(StatefulSet sts, int podId) {
```

Comment:
As I said in the other comment. If I read the code right, you pass the STS in just to get namespace and cluster name. That seems to be very wasteful. Why not just pass the two strings?